generator/template.go

```diff
@@ -195,6 +195,91 @@ function getNotifyEntityArrivalSink<T>(notifyCallback: NotifyStreamEntityArrival
     }
   })
 }
+
+/**
+ * Flattens a deeply nested object
+ *
+ * @param  {Object} deeplyNestedObject
+ * @param  {String} path
+ * @return {Record<string, unknown>}
+ */
+function flattenObject<T extends Record<string, unknown>>(
+  deeplyNestedObject: T,
+  path: string = ""
+): T {
+  return Object.keys(deeplyNestedObject).reduce(
+    (acc: T, key: string): T => {
+      const value = deeplyNestedObject[key];
+
+      const newPath = Array.isArray(deeplyNestedObject)
+        ? path + "." + key
+        : [path, key].filter(Boolean).join(".");
```

Comment:
The `filter(Boolean)` removes the first `.` from the keys in the flattened object. 

For example, a request object like this:
```js
const req = {
  a: 10,
  postReq: {
    b: 11
  },
  c: [23, 25],
  extMsg: {
    d: 12
  }
};
```

Without `filter(Boolean)`:
```js
{
  .a: 10
  .c.0: 23
  .c.1: 25
  .extMsg.d: 12
  .postReq.b: 11
}
```

With `filter(Boolean)`:
```js
{
  a: 10
  c.0: 23
  c.1: 25
  extMsg.d: 12
  postReq.b: 11
}
```

Reply:
This is because the path's not pass down correctly in the first level, then it makes this `filter(Boolean)` required for every nested level, which's not ideal. better just from root level pass down the correct path.

`newPath = path ? [path, key].join(".") : key

Reply:
Ah, I see... I have made a very naive mistake. Thanks for pointing this out 🙏🏼